src/vm/amd64/jithelpers_fastwritebarriers.S

```diff
@@ -244,3 +246,331 @@ PATCH_LABEL JIT_WriteBarrier_SVR64_PatchLabel_CardTable
         mov     byte ptr [rdi + rax], 0FFh
         ret
 LEAF_END_MARKED JIT_WriteBarrier_SVR64, _TEXT
+
+#endif
+
+
+#ifdef FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
+
```

Comment:
I don't think we would see the 32-bit ones getting hit in our tests because we have the high entropy flag on which enables ASLR to reserve the lower addresses. The question is more that if we used the 64-bit ones to handle the 32-bit addresses, is that a big perf loss. I doubt it. As far as the pre and post 64 ones, it's less clear to me - I would think that having pre could give us a slight perf boost in workstation GC (that's hard to detect with microbenchmarks). @PeterSolMS , do you have an opinion?